exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceStreamObservers.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.ocagent;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.stub.StreamObserver;
+import io.opencensus.proto.agent.trace.v1.CurrentLibraryConfig;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceRequest;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceResponse;
+import io.opencensus.proto.agent.trace.v1.UpdatedLibraryConfig;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.config.TraceParams;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+*/
+
+/** Stream observers of OC-Agent Trace Service. */
+final class OcAgentTraceServiceStreamObservers {
+
+  private OcAgentTraceServiceStreamObservers() {}
+
+  private static final Logger logger =
+      Logger.getLogger(OcAgentTraceServiceStreamObservers.class.getName());
+
+  /*
+   * We can use ManagedChannel.getState() to determine the state of current connection.
+   *
+   * 0. For export RPC we hold a reference "exportRequestObserverRef" to a StreamObserver for
+   * request stream, and "exportRequestObserverRef" to a StreamObserver for response stream.
+   * For config RPC we have "currentConfigObserverRef" and "updatedConfigObserverRef". Note that
```

Comment:
Please don't name the streams based on the message names. Use request/response because that is easier to understand. As I commented in a previous comment.

Reply:
The problem is we have two streams and they are in the opposite direction: export is Exporter -> Agent, while config is Agent -> Exporter. Let me think about how to make this more clear.

Reply:
Took a second look at this block, I still think it's necessary to differentiate between the two streams (export and config) since they work in completely different ways. Using generic request/response for both of them sounds confusing to me. Again this may be my personal bias, so if you feel strongly that request/response are easier for others to understand (probably everyone except me), I'm happy to re-phrase it that way.

Reply:
I.e from Exporter's perspective:
- For `export`, Exporter is sending requests and receiving responses.
- For `config`, Exporter is receiving requests and sending responses.

Reply:
You can name the config streams:
configRequestObserverRef
configResponseObserverRef

Reply:
SG, updated.